packages/account/src/predicate/predicate.ts

```diff
@@ -93,6 +96,16 @@ export class Predicate<TInputData extends InputValue[]> extends Account {
         input.predicate = this.bytes;
         // eslint-disable-next-line no-param-reassign
         input.predicateData = this.getPredicateData(policies.length);
+
+        /**
+         * Not sure if this is needed. The only scenario that this could help is if
+         * the user has fetched the predicate resources using the provider `getResourcesToSpend` method,
+         * as the method will not populate the predicate property and the resources are going to be added
+         * as common resources. We should enforce users to always fetch resources using the predicate instance,
+         * as they will need it to populate the predicate data anyway using this method.
+         */
+        // eslint-disable-next-line no-param-reassign
+        input.witnessIndex = 0;
```

Comment:
Should we throw an Error then, suggesting something along the lines of the `witnessIndex` is missing - please ensure you are fetching resources using the predicate instance?